pkg/doc/verifiable/linked_data_proof_test.go

```diff
@@ -78,21 +84,91 @@ func TestLinkedDataProofSignerAndVerifier(t *testing.T) {
 }
 `
 
-	vc, err := NewUnverifiedCredential([]byte(vcJSON))
+	ed25519PubKey, ed25519PrivKey, err := ed25519.GenerateKey(rand.Reader)
 	require.NoError(t, err)
 
-	created, err := time.Parse(time.RFC3339, "2018-03-15T00:00:00Z")
+	vcWithEd25519Proof := prepareVCWithEd25519LDP(t, vcJSON, ed25519PrivKey)
+
+	vcWithEd25519ProofBytes, err := vcWithEd25519Proof.MarshalJSON()
+	require.NoError(t, err)
+
+	ecdsaPrivKey, err := ecdsa.GenerateKey(btcec.S256(), rand.Reader)
 	require.NoError(t, err)
+	vcWithSecp256k1Proof := prepareVCWithSecp256k1LDP(t, vcJSON, ecdsaPrivKey)
 
-	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	vcWithSecp256k1ProofBytes, err := vcWithSecp256k1Proof.MarshalJSON()
 	require.NoError(t, err)
+	require.NotEmpty(t, vcWithSecp256k1ProofBytes)
 
-	signerSuite := ed25519signature2018.New(
+	t.Run("Single signature suite", func(t *testing.T) {
+		verifierSuite := ed25519signature2018.New(
+			suite.WithVerifier(ed25519signature2018.NewPublicKeyVerifier()),
+			suite.WithCompactProof())
+		vcDecoded, _, err := NewCredential(vcWithEd25519ProofBytes,
+			WithEmbeddedSignatureSuites(verifierSuite),
+			WithPublicKeyFetcher(SingleKey(ed25519PubKey, kms.ED25519)))
+		require.NoError(t, err)
+		require.Equal(t, vcWithEd25519Proof, vcDecoded)
+	})
+
+	t.Run("Several signature suites", func(t *testing.T) {
+		verifierSuites := []VerifierSignatureSuite{
+			ed25519signature2018.New(
+				suite.WithVerifier(ed25519signature2018.NewPublicKeyVerifier()),
+				suite.WithCompactProof()),
+			ecdsasecp256k1signature2019.New(
+				suite.WithVerifier(ecdsasecp256k1signature2019.NewPublicKeyVerifier())),
+		}
+
+		vcDecoded, _, err := NewCredential(vcWithEd25519ProofBytes,
+			WithEmbeddedSignatureSuites(verifierSuites...),
+			WithPublicKeyFetcher(SingleKey(ed25519PubKey, kms.ED25519)))
+		require.NoError(t, err)
+		require.Equal(t, vcWithEd25519Proof, vcDecoded)
+
+		pubKeyBytes := elliptic.Marshal(ecdsaPrivKey.Curve, ecdsaPrivKey.X, ecdsaPrivKey.Y)
+		vcDecoded, _, err = NewCredential(vcWithSecp256k1ProofBytes,
+			WithEmbeddedSignatureSuites(verifierSuites...),
+			WithPublicKeyFetcher(func(issuerID, keyID string) (*verifier.PublicKey, error) {
+				return &verifier.PublicKey{
+					Type:  "EcdsaSecp256k1VerificationKey2019",
+					Value: pubKeyBytes,
+					JWK: &jose.JWK{
+						JSONWebKey: gojose.JSONWebKey{
+							Algorithm: "ES256K",
+						},
+						Crv: "secp256k1",
+						Kty: "EC",
+					},
```

Comment:
The [spec](https://w3c-ccg.github.io/ld-cryptosuite-registry/#ecdsa-secp256k1) doesn't specify JWK for `EcdsaSecp256k1VerificationKey2019` key type. Is this a dummy data for unit test or does the code really expect the DIDDoc to have JWK value for this key type?

Reply:
At https://w3c-ccg.github.io/lds-ecdsa-secp256k1-2019/, you can find an example with the JWK. `publicKeyHex` should be supported in the `EcdsaSecp256k1Signature2019` suite, I will create an issue for that.

Reply:
https://github.com/hyperledger/aries-framework-go/issues/1609